psychopy/experiment/components/_base.py

```diff
@@ -498,6 +498,47 @@ def writeParamUpdate(self, buff, compName, paramName, val, updateType,
                 if stopVal in ['', None, -1, 'None']:
                     stopVal = '-1'
                 buff.writeIndented(f"{compName}.setSound({params['sound']}, secs={stopVal}){endStr}\n")
+            elif paramName == 'movie':
+                # we're going to do this for now ...
+                if params['units'].val == 'from exp settings':
+                    unitsStr = "units=''"
+                else:
+                    unitsStr = "units=%(units)s" % params
+
+                if params['backend'].val == 'moviepy':
+                    code = ("%s = visual.MovieStim3(\n" % params['name'] +
```

Comment:
Using `%s` is a bit hard to follow - it's better to do `"%(name)s = visual.MovieStim3(\n"` and then at the end of the whole thing just do `% params`. For things like `unitsStr` you can either add it on manually or use an f string with `{}` so it doesn't conflict with the modulus formatting.